src/ui/components/Timeline/index.js

```diff
@@ -761,6 +766,49 @@ export class Timeline extends Component {
     ];
   }
 
+  onScrollContainerScroll = e => {
+    const { hoverTime, setZoomRegion } = this.props;
+
+    const sensitivity = 200;
+    const scrollTop = e.target.scrollTop;
+    console.log("scrollTop", scrollTop);
+    const newScale = 1 + scrollTop / sensitivity;
+
+    const selection = this.calculateNewSelection(hoverTime, newScale);
+    setZoomRegion({
+      startTime: selection.newStart,
+      endTime: selection.newEnd,
+      mid: selection.newMid,
+      scale: newScale,
+    });
+    // gToolbox.getWebconsoleWrapper().setZoomedRegion(start, end);
+  };
+
+  calculateNewSelection(hover, newScale) {
+    const { recordingDuration, zoomRegion } = this.props;
+    let length = zoomRegion.endTime - zoomRegion.startTime;
+    let mid = zoomRegion.mid || length / 2;
+    let scale = zoomRegion.scale;
+
+    const newLength = (length * scale) / newScale;
+    const newMid = mid + (hover - mid) * ((newScale - scale) / newScale);
+    const newStart = newMid - newLength / 2;
+    const newEnd = newMid + newLength / 2;
+
+    return { newLength, newMid, newStart, newEnd };
+  }
```

Comment:
couple of high level thoughts:

1. it'd be nice if this function returned the zoomRegion type, basically drop the `new` :) 
2. it looks like we don't need to return length
3. it'd be nice if this helper function could live in a util. Given that, i'd probably just pass in the props as well  `calculateNewSelection({hover, newScale, recordingDuration, zoomRegion})`